src/arm7tdmi.rs

```diff
@@ -29,40 +35,90 @@ impl Cpu for Arm7tdmi {
         op_code
     }
 
-    fn decode(&self, op_code: Self::OpCodeType) -> (Condition, InstructionKind) {
+    fn decode(&self, op_code: Self::OpCodeType) -> (Condition, Self::InstructionType) {
+
         let condition = (op_code >> 28) as u8; // bit 31..=28
-        let instruction = if (op_code & 0x0A_00_00_00) != 0 {
-            if (op_code & 0x01_00_00_00) != 0 {
-                InstructionKind::BranchLink
-            } else {
-                InstructionKind::Branch
-            }
-        } else {
-            todo!()
-        };
         println!("condition -> {:x}", condition);
+
+        let res_decode = ArmModeInstruction::get_instruction(op_code);
+        if res_decode.is_err() {
```

Comment:
Probably we need to start to handle error with `Result` or directly `panic
` if we prefer

Reply:
Yes, this is only a draft to manage the instructions not implemented yet.